src/traces/scattergl/index.js

```diff
@@ -47,116 +48,73 @@ function calc(gd, trace) {
     var xa = Axes.getFromId(gd, trace.xaxis);
     var ya = Axes.getFromId(gd, trace.yaxis);
     var subplot = fullLayout._plots[trace.xaxis + trace.yaxis];
+    var count = trace._length;
+    var count2 = count * 2;
     var stash = {};
+    var i, xx, yy;
 
+    var x = xa.makeCalcdata(trace, 'x');
+    var y = ya.makeCalcdata(trace, 'y');
 
-    var x = xaxis.type === 'linear' ? trace.x : xaxis.makeCalcdata(trace, 'x');
-    var y = yaxis.type === 'linear' ? trace.y : yaxis.makeCalcdata(trace, 'y');
```

Comment:
**IMPORTANT** with typed array support in mind (-> https://github.com/plotly/plotly.js/pull/2388), I made all traces pass through `makeCalcdata` unlike previously where x/y array on linear axes bypassed it. Please note that `makeCalcdata` creates a new array (i.e. `x` isn't the same as `trace.x`), unless `trace.x` is a typed array. So memory-conscious user should switch to using typed arrays.

Note that at `1e6` on my setup, one makeCalcdata` call clocks in at roughly `25ms`. So using typed arrays can save about `50ms`

Reply:
Even with the extra arrays, memory consumption appears fine. I wouldn't having @dfcreative double-check though.

Reply:
I guess there are situations we accept now that would have broken previously (but only on linear axes) - the "junk" characters stripped by [`cleanNumber`](https://github.com/plotly/plotly.js/blob/master/src/lib/clean_number.js) - worth a test case?

Reply:
Yes. Good call!

Reply:
Added mock in c15722f.

On master, the new mock would have looked like:

![image](https://user-images.githubusercontent.com/6675409/36763611-bbf1fa42-1bf6-11e8-844f-0c9b14b99082.png)